src/history/suborchestrationinstancefailedevent.ts

```diff
@@ -16,6 +16,18 @@ export class SubOrchestrationInstanceFailedEvent extends HistoryEvent {
             options.timestamp,
         );
 
+        if (options.taskScheduledId === undefined) {
+            throw new Error("SubOrchestrationInstanceFailedEvent needs a task scheduled id provided.");
+        }
+
+        if (options.reason === undefined) {
+            throw new Error("SubOrchestrationInstanceFailedEvent needs a reason provided.");
```

Comment:
Do we require a reason? I don't expect we needed that in .NET.

Reply:
Good call.